applications/asset_tracker/src/cloud_codec/service_info.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include "service_info.h"
+
+#define SERVICE_INFO_JSON_NAME "serviceInfo"
+#define SENSORS_JSON_NAME "sensors"
+#define FOTAS_JSON_NAME "fota_v1"
+
+static int add_array_obj(const char * const items[], const u32_t item_cnt,
+			 const char * const item_name, cJSON * const obj)
+{
+	if (!obj || !item_name) {
+		return -EINVAL;
+	}
+
+	cJSON * array = cJSON_CreateArray();
+
+	if (!array) {
+		return -ENOMEM;
+	}
+
+	for (u32_t cnt = 0; items && (cnt < item_cnt); ++cnt) {
+		if (items[cnt]) {
+			cJSON_AddItemToArray(array,
+					     cJSON_CreateString(items[cnt]));
+		}
+	}
+
+	cJSON_AddItemToObject(obj, item_name, array);
```

Comment:
Just to be sure - it is intended to add an empty array in case `items` were NULL?

Reply:
that was my intent, i'll check and see if the front end has a preference

Reply:
adding a null value is preferred instead of the empty array.
updated.